skepticoin/datatypes.py

```diff
@@ -1,85 +1,95 @@
+from __future__ import annotations
+
+from io import BytesIO
 import struct
+from typing import Any, List
 
-from .hash import sha256d
-from .humans import human
-from .params import CHAIN_SAMPLE_TOTAL_SIZE
-from .serialization import (
+from skepticoin.hash import sha256d
+from skepticoin.humans import human
+from skepticoin.params import CHAIN_SAMPLE_TOTAL_SIZE
+from skepticoin.serialization import (
     Serializable,
     safe_read,
     stream_deserialize_list,
     stream_deserialize_vlq,
     stream_serialize_list,
     stream_serialize_vlq,
 )
-from .signing import PublicKey, SignableEquivalent, Signature
+from .signing import PublicKey, SECP256k1Signature, SignableEquivalent, Signature
 
 
 class OutputReference(Serializable):
     """Refer an output by its transaction hash and index into its list of outputs."""
 
-    def __init__(self, hash, index):
+    def __init__(self, hash: bytes, index: int):
         if not len(hash) == 32:
             raise ValueError("OutputReference hash must be 32 bytes.")
 
         if not (0 <= index <= 0xFFFFFFFF):
-            raise ValueError("OutputReference index out of range." % index)
+            raise ValueError("OutputReference index %d is out of range." % index)
```

Comment:
Another good catch (either by you or by your tooling)

Reply:
Either black or mypy caught this one.